### PR TITLE
pytensor 2.23.0

### PR DESCRIPTION
**Pull request by psteyer**
pytensor 2.23.0

**Destination channel:** defaults

### Links

- [PKG-5198](https://anaconda.atlassian.net/browse/PKG-5198) 
- [Upstream repository](https://github.com/pymc-devs/pytensor)
- [Upstream changelog/diff](https://github.com/pymc-devs/pytensor/compare/rel-2.13.1...rel-2.23.0)

### Explanation of changes:

- Updated entry-point

[PKG-5198]: https://anaconda.atlassian.net/browse/PKG-5198?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -86,4 +86,4 @@ extra:
   skip-lints:
     - host_section_needs_exact_pinnings
     - python_build_tool_in_run
-    - documentation_specifies_language
\ No newline at end of file
+    - documentation_specifies_language
```
Comment (ViridianMelody) [inline recipe/meta.yaml]:
```suggestion
```

Should be able to just adjust doc_url to https://pytensor.readthedocs.io rather than skipping the lint.

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -59,7 +60,6 @@ test:
     - check-for-warnings.py
     - allowed-warnings-base.txt
   commands:
-    - pytensor-cache help
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:62]: We should keep this.

Comment (psteyer) [reply recipe/meta.yaml:62]: Woops! I thought I added it back in after changing the entry point. Thanks for the catch!

**Comment by ifitchet** [comment]
Do you need `setuptools` in `requirements.run`?

**Comment by JeanChristopheMorinPerso** [comment]
>Do you need setuptools in requirements.run?

Yes we do. It's used in `pytensor.link` to, read notes, dynamically generate C/JAX/Numba code (evaluate expressions faster). I don't what what that does, but it's used for that.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**